### PR TITLE
issue #28896 - allow for extensions defining their own wftypes

### PR DESCRIPTION
**Pull request by gilmoskowitz**
@rkannadevara - please try to reproduce bug 28896 with this updated version of the `createwf()` function. It should give more information that may help find the real source of the bug.

### CODE REVIEW

**Comment by sindhupriyaav** [comment]
Performed UAT: Observed that on selecting to release work order, 'Error Retrieving Work Order Information' dialog with dblog error is displayed. Please refer the below db log error , screenshot.

workorderrelease
Db Log error:
ERROR:  Cannot find source table needed to generate workflow [xtuple: createwf, -3, wo, WO]
CONTEXT:  SQL statement "SELECT xt.createwf(TG_TABLE_NAME, NEW)"
PL/pgSQL function createwf_after_insert() line 5 at PERFORM
SQL statement "UPDATE wo
  SET wo_status='R'
  WHERE ((wo_status='E')
   AND (wo_id=pWoid))"
PL/pgSQL function releasewo(integer,boolean) line 9 at SQL statement
(P0001) QPSQL: Unable to create query

The Query:
SELECT        releaseWo(wo_id, false)        FROM wo, itemsite, plancode WHERE ((wo_itemsite_id=itemsite_id)  AND  (itemsite_plancode_id=plancode_id)  AND  (wo_status='E')  AND  (wo_startdate<=:_1_ ));

Bound Values:
:_1_: 2017-02-13

File: database/source/xt/functions/createwf.sql [hunk 1]
```diff
@@ -1,149 +1,134 @@
...
 
-      PERFORM xt.workflow_inheritsource('xt.' || _source_model, _workflow_class, _item_uuid, _parent_id, _order_id);
+    ELSE
+      RAISE EXCEPTION 'Improper table supplied to createwf function [xtuple: createwf, -2, %]', tg_table_name;
+    END IF;
+
+    SELECT wftype_src_tblname INTO _source_model
+      FROM xt.wftype
+     WHERE wftype_code = _wftypecode;
+    IF _source_model IS NOT NULL THEN
```
Comment (gilmoskowitz) [inline database/source/xt/functions/createwf.sql]: This conditional is the crux of the bug fix - only `workflow_inheritsource()` if the current extensions support it. Bug 28896 was caused by the distribution edition [not having a wftype record for `wo`](https://github.com/xtuple/private-extensions/blob/4_11_x/source/manufacturing/database/source/xt/tables/wftype.sql).

Comment (bixwilson) [inline database/source/xt/functions/createwf.sql]: Yes, this looks like a reasonable fix. Thanks.

**Comment by sindhupriyaav** [comment]
Performed UAT: Observed that on selecting to release work order, 'Error Retrieving Work Order Information' dialog with dblog error is displayed. Please refer the below db log error , screenshot.

 
distempty

DBLog Error:

TypeError: Cannot call method 'split' of null
DETAIL:  workflow_inheritsource() LINE 30:   namespace = source_model.split(".")[0];
CONTEXT:  SQL statement "SELECT xt.workflow_inheritsource('xt.' || source_model, workflow_class, item_uuid, parent_id, _order_id)"
PL/pgSQL function createwf(text,anyelement) line 132 at PERFORM
SQL statement "SELECT xt.createwf(TG_TABLE_NAME, NEW)"
PL/pgSQL function createwf_after_insert() line 5 at PERFORM
SQL statement "UPDATE wo
SET wo_status='R'
WHERE ((wo_status='E')
AND (wo_id=pWoid))"
PL/pgSQL function releasewo(integer,boolean) line 9 at SQL statement
(XX000) QPSQL: Unable to create query
SELECT        releaseWo(wo_id, false)        FROM wo, itemsite, plancode WHERE ((wo_itemsite_id=itemsite_id)  AND  (itemsite_plancode_id=plancode_id)  AND  (wo_status='E')  AND  (wo_startdate<='2017-01-15' )  AND  (itemsite_warehous_id=1 ));
Fri Dec 16 15:00:20 2016 ERROR:  TypeError: Cannot call method 'split' of null
DETAIL:  workflow_inheritsource() LINE 30:   namespace = source_model.split(".")[0];
CONTEXT:  SQL statement "SELECT xt.workflow_inheritsource('xt.' || source_model, workflow_class, item_uuid, parent_id, _order_id)"
PL/pgSQL function createwf(text,anyelement) line 132 at PERFORM
SQL statement "SELECT xt.createwf(TG_TABLE_NAME, NEW)"
PL/pgSQL function createwf_after_insert() line 5 at PERFORM
SQL statement "UPDATE wo
SET wo_status='R'
WHERE ((wo_status='E')
AND (wo_id=pWoid))"
PL/pgSQL function releasewo(integer,boolean) line 9 at SQL statement
(XX000) QPSQL: Unable to create query
SELECT        releaseWo(wo_id, false)        FROM wo, itemsite, plancode WHERE ((wo_itemsite_id=itemsite_id)  AND  (itemsite_plancode_id=plancode_id)  AND  (wo_status='E')  AND  (wo_startdate<='2017-01-15' )  AND  (itemsite_warehous_id=1 ));

**Pull request merged**